lib/vintage_net_mobile/modem/automatic/discovery.ex

```diff
@@ -0,0 +1,58 @@
+defmodule VintageNetMobile.Modem.Automatic.Discovery do
+  @moduledoc false
+
+  # A process for trying to discover a the type of modem and some basic
+  # information to allow dynamic configuration items
+
+  use GenServer
+
+  alias Circuits.UART
+
+  @quectel_vender_id 0x2C7C
+
+  @modems %{
+    {@quectel_vender_id, 0x0125} => VintageNetMobile.Modem.QuectelEC25,
+    {@quectel_vender_id, 0x0296} => VintageNetMobile.Modem.QuectelBG96
+  }
+
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(args) do
+    modem = detect_modem()
+
+    {:ok, %{modem: modem, raw_config: args[:raw_config]}, {:continue, :configure_vintage_net}}
+  end
+
+  @impl GenServer
+  def handle_continue(:configure_vintage_net, state) do
+    config =
+      put_in(state, [:raw_config, :source_config, :vintage_net_mobile, :modem], state.modem)
+
+    VintageNet.configure(state.raw_config.ifname, %{type: VintageNet, vintage_net_mobile: config},
+      persist: false
+    )
+
+    {:noreply, state}
+  end
+
+  defp detect_modem() do
+    UART.enumerate()
```

Comment:
With no modem attached, this will continually enumerate very quickly. Is there a performance consideration in that case? Should we handle retrying more gracefully?

Reply:
Yeah, I kicked the tires around with backing off the enumeration. I kinda took the road of getting the interface up as fast as possible. My basic assumption is VintageNet will not start the interface if the hardware is not present, but I need to verify that.